packages/botonic-plugin-flow-builder/src/action/index.tsx

```diff
@@ -43,11 +43,18 @@ export class FlowBuilderAction extends React.Component<FlowBuilderActionProps> {
     ) as FlowHandoff
     if (handoffContent) await handoffContent.doHandoff(request)
 
-    const renderContents = contents.filter(content =>
+    const contentsWithoutHandoff = contents.filter(content =>
       content instanceof FlowHandoff ? false : true
     )
 
-    return { contents: renderContents }
+    const contentsWithVariables = contentsWithoutHandoff.map(content => {
+      if (content instanceof FlowText) {
+        content.text = content.replaceVariables(request.session.user.extra_data)
+      }
```

Comment:
Change: I think this should be called inside the `fromHubtypeCMS` method of the `FlowText` class. As @AinaVendrell all these logic should be placed and executed inside the `FlowText` class, the action should not be aware of this logic.

To make it work you will have to pass the request to the `FlowBuilderApi` class, maybe you can add it as an argument of the constructor.